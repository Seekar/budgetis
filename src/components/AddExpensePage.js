import React from "react";

const AddExpensePage = () => {
  return <div>This is from my CreatePage</div>;
};

export default AddExpensePage;
