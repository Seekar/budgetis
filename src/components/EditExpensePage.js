import React from "react";

const EditExpensePage = () => {
  return <div>This is from my EditPage</div>;
};

export default EditExpensePage;
