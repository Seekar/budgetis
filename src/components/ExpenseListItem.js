import React from "react";
import { connect } from "react-redux";
import { removeExpense } from "../actions/expensesActions";
import { Link } from "react-router-dom";

class ExpenseListItem extends React.Component {
  onRemove = () => {
    this.props.removeExpense(this.props.expense.id);
  };

  render() {
    const { id, description, note, amount, createdAt } = this.props.expense;
    return (
      <tr>
        <td>{description}</td>
        <td>{note}</td>
        <td>{amount}</td>
        <td>{createdAt}</td>
        <td>
          <button onClick={this.onRemove}>Remove</button>
          <Link to={`/edit/${id}`}>
            <button>Edit</button>
          </Link>
        </td>
      </tr>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  removeExpense: data => dispatch(removeExpense(data))
});

ExpenseListItem.propTypes = {
  //name: PropTypes.string
};

export { ExpenseListItem };

export default connect(undefined, mapDispatchToProps)(ExpenseListItem);
