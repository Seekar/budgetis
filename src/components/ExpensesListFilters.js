import React, { Component } from "react";
import { connect } from "react-redux";
import {
  sortByAmountFilter,
  sortByDateFilter,
  setTextFilter,
  setStartDateFilter,
  setEndDateFilter
} from "../actions/filtersActions";

import { DateRangePicker } from "react-dates";

class ExpensesListFilters extends Component {
  constructor(props) {
    super(props);
    this.state = { focusedInput: null };
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = focusedInput => {
    this.setState(() => ({ focusedInput }));
  };
  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = e => {
    if (e.target.value === "date") {
      this.props.sortByDate();
    } else if (e.target.value === "amount") {
      this.props.sortByAmount();
    }
  };

  render() {
    return (
      <div>
        Search By :{" "}
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        ></input>{" "}
        Sort By :{" "}
        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId="start_date_id"
          endDate={this.props.filters.endDate}
          endDateId="end_date_id"
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.focusedInput}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
  }
}

ExpensesListFilters.propTypes = {
  //name: PropTypes.string
};

const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDateFilter()),
  sortByAmount: () => dispatch(sortByAmountFilter()),
  setStartDate: startDate => dispatch(setStartDateFilter(startDate)),
  setEndDate: endDate => dispatch(setEndDateFilter(endDate))
});

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export { ExpensesListFilters };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExpensesListFilters);
