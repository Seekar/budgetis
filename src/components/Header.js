import React from "react";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";

const Header = () => {
  return (
    <header>
      <h1>Expensify</h1>
      <NavLink to="/" activeClassName="is-active" exact={true}>
        DashBoard
      </NavLink>

      <NavLink to="/create" activeClassName="is-active">
        {" "}
        Add Expense
      </NavLink>

      {/*       <NavLink to="/edit" activeClassName="is-active">
        {" "}
        Edit Expense
      </NavLink> */}

      <NavLink to="/help" activeClassName="is-active">
        {" "}
        Help
      </NavLink>
    </header>
  );
};

Header.propTypes = {
  //name: PropTypes.string
};

export default Header;
