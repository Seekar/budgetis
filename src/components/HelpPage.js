import React from "react";

const HelpPage = () => {
  return <div>This is from my HelpPage</div>;
};

export default HelpPage;
