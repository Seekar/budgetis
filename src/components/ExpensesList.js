import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import getVisibleExpenses from "../selectors/expensesSelector";

const ExpensesList = props => {
  const { expenses } = props;

  return (
    <div>
      <h1>Expenses List ({`${expenses.length} items`})</h1>
      {expenses && (
        <table>
          <thead>
            <tr>
              <th>Description</th>
              <th>note</th>
              <th>Amount</th>
              <th>Created At</th>
              <th>Actions</th>
            </tr>
          </thead>
          <tbody>
            {expenses.map(exp => (
              <ExpenseListItem key={exp.id} expense={exp}></ExpenseListItem>
            ))}
          </tbody>
        </table>
      )}
    </div>
  );
};

const mapStateToProps = state => {
  const { expenses, filters } = state;
  return {
    expenses: getVisibleExpenses(expenses, filters)
  };
};

export default connect(mapStateToProps)(ExpensesList);
