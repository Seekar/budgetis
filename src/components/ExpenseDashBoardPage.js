import React from "react";
import ExpensesList from "./ExpensesList";
import ExpensesListFilters from "./ExpensesListFilters";

const ExpenseDashboardPage = () => {
  return (
    <div>
      <ExpensesListFilters />
      <ExpensesList />
    </div>
  );
};

ExpenseDashboardPage.propTypes = {
  //name: PropTypes.string
};

export default ExpenseDashboardPage;
