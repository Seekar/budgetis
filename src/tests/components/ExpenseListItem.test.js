import React from "react";
import { shallow } from "enzyme";
import { ExpenseListItem } from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

let removeExpense, history, wrapper;

beforeEach(() => {
  removeExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <ExpenseListItem
      removeExpense={removeExpense}
      history={history}
      expense={expenses[2]}
    />
  );
});

test("should render ExpensesListItem with expenses", () => {
  const wrapper = shallow(<ExpenseListItem expense={expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should handle removeExpense", () => {
  wrapper
    .find("button")
    .first()
    .simulate("click");
  expect(removeExpense).toHaveBeenLastCalledWith(expenses[2].id);
});
