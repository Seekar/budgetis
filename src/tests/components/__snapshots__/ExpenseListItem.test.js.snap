// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesListItem with expenses 1`] = `
<tr>
  <td>
    Gum
  </td>
  <td />
  <td>
    195
  </td>
  <td>
    0
  </td>
  <td>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
    <Link
      to="/edit/1"
    >
      <button>
        Edit
      </button>
    </Link>
  </td>
</tr>
`;
